### PR TITLE
Fix GCS credentials when no endpoint specified

### LINKED ISSUE #137: GCS broken due to wrong HTTP client
**Describe the bug**
GCS errors with 403:

```<?xml version='1.0' encoding='UTF-8'?><Error><Code>AccessDenied</Code><Message>Access denied.</Message><Details>Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object.</Details></Error>```

After a lot of tracing, I realized that if `GCS_ENDPOINT` isn't set, the Google SDK is called with `option.WithHTTPClient()` using a custom HTTP client that disables TLS verification. I'm not sure why this breaks anything, but it does. Perhaps there's something that doesn't send auth headers over insecure connections?

**To Reproduce**
Try using the GCS backend without `GCS_ENDPOINT`.

### PR DESCRIPTION
**Pull request by atombender**
### Description

Fixes #137 by not overriding the HTTP client when not necessary.

Without this, GCS errors with 403:

```
<?xml version='1.0' encoding='UTF-8'?><Error><Code>AccessDenied</Code><Message>Access denied.</Message><Details>Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object.</Details></Error>
```

This happens when `GCS_ENDPOINT` isn't set, and the Google SDK is called with `option.WithHTTPClient()` using a custom HTTP client that disables TLS verification. 

### Checklist

- [x] Read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] Read the [**CODE OF CONDUCT**](CODE_OF_CONDUCT.md) document.
- [x] Add tests to cover changes. [Not possible since this only happens with TLS.]
- [x] Ensure your code follows the code style of this project.
- [ ] Ensure CI and all other PR checks are green OR
    - [x] Code compiles correctly.
    - [x] All new and existing tests passed.
- [x] Add your changes to `Unreleased` section of [CHANGELOG](CHANGELOG.md).

### CODE REVIEW

File: storage/backend/gcs/gcs.go [hunk 1]
```diff
@@ -35,16 +35,16 @@ func New(l log.Logger, c Config) (*Backend, error) {
...
+		if !strings.HasPrefix(c.Endpoint, "https://") { // This is not settable from outside world, only used for mock tests.
+			opts = append(opts, option.WithHTTPClient(&http.Client{Transport: &http.Transport{
+				// ignore unverified/expired SSL certificates for tests.
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, //nolint:gosec
+			}}))
+		}
 	}
 
-	setAuthenticationMethod(l, c, opts)
+	opts = setAuthenticationMethod(l, c, opts)
```
Comment (kakkoyun) [inline storage/backend/gcs/gcs.go:47]: 🙈

**Review by kakkoyun — DISMISSED** [review]
Thanks a lot for your contribution! Another great PR! Thanks a lot for fixing those mistakes ❤️

**Review by kakkoyun — APPROVED** [review]

**Comment by danihodovic** [comment]
@kakkoyun could you please update the Docker hub image to include this patch?

**Comment by danihodovic** [comment]
Goreleaser has deprecated building Docker images, so you will have to change that too.

**Comment by kakkoyun** [comment]
Hey @danihodovic, would you be up for handling it? We always welcome to new contributors :hugs:

**Comment by danihodovic** [comment]
I would, but I'm on a busy schedule for the next month. I've resolved the blocker for myself by modifying the Dockerfile, building manually and pushing to `danihodovic/drone-cache`. The image is rather large, but it works for me.


```
FROM golang:1.14.4-alpine3.12 AS builder
RUN apk add --update --no-cache ca-certificates tzdata && update-ca-certificates
COPY . /app/
WORKDIR /app/
RUN go build
RUN go install
ENTRYPOINT ["drone-cache"]
```

**Comment by seandtaber** [comment]
> @danihodovic We should release a new version. It's been a while. @seandtaber What do you think? Can someone take look at it on your side?

We can take a look at this issue and will respond here when we identify the issue.

**Pull request merged**